packages/contracts-bedrock/contracts/periphery/OpStackChainRegistry.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/**
+ * @title  OpStackChainRegistry
+ * @notice OpStackChainRegistry acts as a way to register and query
+ *         contract addresses for an OP Stack chain
+ */
+contract OpStackChainRegistry {
+
+    error OnlyDeploymentAdmin();
+
+    /**
+     * @notice Struct representing a deployment
+     *         on an OP Stack chain
+     */
+    struct Deployment {
+        string deploymentName;
+        address deploymentAdmin;
+    }
+
+    /**
+     * @notice Struct representing a deployment entry with
+     *         a contract's name and associated address
+     */
+    struct DeploymentEntry {
+        string entryName;
+        address entryAddress;
+    }
+
+    /**
+     * @notice Mapping of deployment names to deployment admins
+     */
+    mapping(string => address) public deployments;
+
+    /**
+     * @notice Mapping of deployments to their chain registry
+     */
+    mapping(string => mapping(string => address)) public registry;
+
+    /**
+     * @notice Claims a deployment
+     *
+     * @param deployment The deployment to claim
+     */
+    function claimDeployment(Deployment calldata deployment) public {
```

Comment:
You should generally be able to recover the state of a smart contract by applying the ordered set of events. No event here. This way of thinking about things enforces thinking of smart contracts as state machines